compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts

```diff
@@ -222,6 +222,14 @@ function writeDependencyToInstructions(
        */
       break;
     }
+    if (path.property === 'current') {
+      /*
+       * Prune ref.current accesses. This may over-capture for non-ref values with
+       * a current property, but that's fine.
+       */
+      currValue.effect = Effect.Freeze;
+      break;
```

Comment:
i haven't followed the implementation here, but why are we changing the effects here? if we're pruning i'd expect that to be operating on the dependency path but not changing effects, which should have already been inferred.

Reply:
This is just unnecessary and I have no idea how that line ended up here 
you're fast on draft diffs :P

Reply:
@josephsavona  The freeze here is redundant with a later freeze (before the function return). The reason we're manually setting a `Effect.Freeze` is because this pass inserts new instructions with new Places.

We could set the effect to be `Effect.ConditionallyMutate`, but inserted auto-deps really has freeze effects. This really should be a no-op with `enableTransitivelyFreezeFunctionExpressions`, which already freezes captured dependencies. One thing we could do is validate `enableTransitivelyFreezeFunctionExpressions` is set with `inferEffectDependencies`.